mantidproject/mantid
# Move SANS gui property handling to a separate class

**Description of work.**
sans_data_processor_gui and add_runs_page, two gui pages within the ISIS SANS interface, use qt properties to set default properties on loading of the gui. The functionality to achieve this has been moved to a separate class to clean up the code, and to make it easier for any other gui page to set and grab properties in the future

This PR also moves a config observer from the add_runs_presenter to run_tab_presenter, as we no longer need it in the add_runs_presenter

**To test:**
1. Interfaces -> SANS -> SANS ISIS
2. On the main gui, select an output mode (one of Memory, File, Both) and toggle Save Can
3. In the add runs page, select an output folder
4. Close and re-open the gui. The settings you just changed should have been remembered

Fixes partially #24529

*This does not require release notes* because **internal code change**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
